jaxrs/src/main/java/com/cerner/beadledom/jaxrs/exceptionmappers/ThrowableExceptionMapper.java

```diff
@@ -0,0 +1,51 @@
+package com.cerner.beadledom.jaxrs.exceptionmappers;
+
+import static javax.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
+
+import com.cerner.beadledom.jaxrs.models.JsonError;
+
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.ext.ExceptionMapper;
+import javax.ws.rs.ext.Provider;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Exception Mapper for all unhandled Exceptions thrown by the service.
+ *
+ * @author Brian van de Boogaard
+ * @author Cal Fisher
+ */
+@Provider
+public class ThrowableExceptionMapper implements ExceptionMapper<Throwable> {
+
+  private static final Logger logger = LoggerFactory.getLogger(ThrowableExceptionMapper.class);
+
+  /**
+   * Generates a Response object with a Status of 500, Content-Type 'application/json', and a
+   * JsonError entity containing details about the unhandled exception in Json format.
+   *
+   * @param exception the Throwable exception that was not handled
+   * @return a json response with status 500
+   */
+  @Override
+  public Response toResponse(Throwable exception) {
+
+    logger.error("Unhandled Exception Occurred.", exception);
+
+    return Response
+        .status(Response.Status.INTERNAL_SERVER_ERROR)
+        .entity(createJsonError())
+        .type(MediaType.APPLICATION_JSON)
+        .build();
+  }
+
+  private static JsonError createJsonError() {
+    return JsonError.builder()
```

Comment:
How do people feel about just inlining this above?

Reply:
Do you mean removing this method and putting it in the `.entity()` call above?

Reply:
@johnlcox yes.

Reply:
Moved this inline with the `.entity()` call: https://github.com/cerner/beadledom/pull/45/commits/ec2febdf850acc1686b13715c230407d77733959#diff-8fa995d7fc0ffa3c90069a9fc131c59bR46